.github/workflows/DragonflyBSD.yml

```diff
@@ -0,0 +1,45 @@
+name: DragonflyBSD
+on:
+  push:
+    branches:
+      - '*'
+    paths:
+      - '**.c'
+      - '**.h'
+      - '**.am'
+      - '**.m4'
+      - 'cmake/*'
+      - 'extra/*'
+      - '.github/workflows/DragonflyBSD.yml'
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - '**.c'
+      - '**.h'
+      - '**.am'
+      - '**.m4'
+      - 'cmake/*'
+      - 'extra/*'
+      - '.github/workflows/DragonflyBSD.yml'
+
+
+
+jobs:
+  DragonflyBSD:
+    runs-on: macos-12
+    steps:
+    - uses: actions/checkout@v2
+    - uses: vmactions/dragonflybsd-vm@v0
+      with:
+        prepare: |
+          pkg install -y   mbedtls cmake
+        usesh: true
+        run: |
+          mkdir build
+          cd build
+          cmake ..
+          make && make verify

```

Comment:
It is better to split building and testing:

### building

```sh
mkdir -p build
cd build
cmake ..
cmake --build .
```

### testing

```sh
JOBS=1
export CTEST_PARALLEL_LEVEL=$JOBS
export CTEST_OUTPUT_ON_FAILURE=1
cmake --build . --target verify
```

You can take a look at macos example on CI - https://github.com/libevent/libevent/blob/master/.github/workflows/build.yml#L573-L665

It is OK, to not have extended build matrix for now I guess, so building everything w/o any options will be enough